### PR TITLE
Plugin sdk

### PR DESCRIPTION
**Pull request by seut**

### CODE REVIEW

File: docs/plugins.txt [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+See our `example crate plugin`_ for details about that.
+
+.. _plugins_crate_dep:
+
+Plugin Crate Dependency
+=======================
+
+In order to develop a plugin against a crate release, a dependency to
+the crate's single JAR must be set up.
```
Comment (dobe) [inline docs/plugins.txt]: crate's server libraries

File: core/build.gradle [hunk 2]
```diff
@@ -12,6 +12,7 @@ configurations {
 dependencies {
     compile project(':es')
     compile 'commons-codec:commons-codec:1.9'
+    compile 'org.apache.xbean:xbean-finder:4.2'
```
Comment (dobe) [inline core/build.gradle]: notice

File: docs/plugins.txt [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+.. _plugins_loading:
+
+Plugin Loading
+==============
+
+Loading of plugins is done by crate for searching all class path
+element resources for a ``META-INF/services/io.crate.Plugin`` file.
+
+Inside this file just one line is allowed defining the FQN class name
```
Comment (dobe) [inline docs/plugins.txt]: full qualified class name

File: docs/plugins.txt [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+which is implementing the `Plugin interface`_. This is almost the same
+like you may know from Java's ServiceLoader.
+
+Constructor with ``Settings`` argument
+--------------------------------------
```
Comment (dobe) [inline docs/plugins.txt]: isnt there a newline required?

**Comment by dobe** [comment]
otherwise ack

File: docs/plugins.txt (hunk 4 above)
Comment (seut) [inline docs/plugins.txt]: no this isn't required by reSt but I'll add one

**Pull request merged**